### ISSUE TITLE
featureContext returns null value for variant

### ISSUE THREAD

**Issue by testifyqa** [TotalTest/SpecFlow.Contrib.Variants#4]
Hi there,

I am trying to use this to get a variant of a feature tag on one of my specflow tests but it keeps giving me null.

Using SpecFlow.NUnit 3.5.5 and SpecFlow.Contribs.Variants 3.5.5

I have an example feature like

`@Variant:Android_HighSpec`
`@Variant:iOS_HighSpec`
`@Variant:Mac_HighSpec`
`@Variant:Windows_HighSpec`
`Feature: Adding Numbers`

`Scenario: Add two numbers`
      `Given I have the calculator open`
      `When I add 2 + 3`
      `Then the result is 5`

Then in my FeatureHooks I have something like...

`[BeforeFeature(Order = 0)]`
  `public static void BeforeFeature(FeatureContext featureContext, IObjectContainer container)`
      `{`
           ` // hooks code i want to run...`
            `featureContext.TryGetValue("Variant", out string variant);`
      `}`
 

variant always returns null though no matter what :(

I have set my specflow.json accordingly too.

I just tested same setup but with scenarioContext instead of featureContext and scenarioContext correctly got the variant value, but featureContext does not.

using .net core 3.1

**Comment by TotalTest** [comment]
Hi, although you are able to set 'variant' tags at the feature level, the intention was to always treat these scenario variants and not feature variants. i.e. the scenarios are run per variant. The way it's done under the hood to achieve this directly works with scenarios and would be a complete overhaul if we had to support variant features. The tagging of the features is just to simplify tagging all individual scenarios.

There's also potential limitations on the unit test frameworks this package supports, for example if we're talking about a simple unit test then MsTest allows a DataRow attribute on a method (scenario) and you could drive variants that way, but you can't do the same at the class level (feature).

Feel free to fork the repo if you think there's a way to achieve your need. But on my analysis this won't be possible.

**Comment by testifyqa** [comment]
Thank you for your response.

I had a quick look at the src code and from what I could see I think I agree with your analysis.

I used BeforeFeature hooks for initial Appium driver instantiations along with isFullReset=false and isNoReset=true capabilities set.

I was looking at using SpecFlow+ Runner along with Targets but there are issues getting it to work in any IDE on macOS which is why I switched to NUnit.

I guess due to this, I will try and move my Feature hooks to the scenario level.

Anyways, cheers for the explanation.
